### ISSUE TITLE
Cannot read property 'duration' of undefined

### ISSUE THREAD

**Issue by jhdevuk** [travisjeffery/mocha-teamcity-reporter#49]
I've just noticed the following error being thrown when running our client side tests in TeamCity:
```
[12:50:27][Step 3/4] Cannot read property 'duration' of undefined
[12:50:27]
[Step 3/4] TypeError: Cannot read property 'duration' of undefined
    at Runner.<anonymous> (node_modules\mocha-teamcity-reporter\lib\teamcity.js:127:52)
[12:50:27][Step 3/4] C:\BuildAgent\work\2d7908fc7e0f07c9\WhiteLabelFrontEndTests\node_modules\mocha-teamcity-reporter\lib\teamcity.js:127
[12:50:27][Step 3/4] 		log(formatString(SUITE_END, 'mocha.suite', stats.duration, flowId));
```
Nothing's changed with the setup or package versions.

Any idea why the `stats` object would be undefined and what might be causing this?
We're using version `2.4.0`.

Many thanks

**Comment by yairhaimo** [comment]
This is because mocha 6 was released:
https://github.com/mochajs/mocha/commit/42303e2acba217af554294b1174ee53b5627cc33

**Comment by njumbo** [comment]
Hey dear contributors, on our side we also needed to remove the reporter because of this issue. Is there a way to refer to a version of mocha that is lower than 6.0 so we can keep using the reporter? :) 

Thanks

**Comment by jamie-sherriff** [comment]
Hey all, i will have a look into this but i suspect it might be https://github.com/mochajs/mocha/issues/2819

@jhdevuk  Could you please get me a simplified version of your test that breaks? so i can add it to my own test suite so i know what i am doing will solve your issue as i don't have any existing tests that do break like that with mocha 6.


@njumbo you should be able to lock your package.json/package-lock.json to mocha@5 like so
```
  "dependencies": {
    "mocha": "^5.2.0"
  },
```
or do a npm install like
```
npm install mocha@5
```

**Comment by shahata** [comment]
Locking `package.json` will not help, it will only work when manually fiddling with `package-lock.json`. Anyway, I think it is not a good idea that this package depends on mocha, it should be in `peerDependencies` instead.

**Comment by jamie-sherriff** [comment]
@shahata ah thanks good point!

**Comment by roymiloh** [comment]
@jamie-sherriff maybe `>=3.5.0 <6` for `mocha`?

**Comment by jamie-sherriff** [comment]
@roymiloh was thinking that but i already have a workaround for mocha 6 i was thinking of doing the following then:

1. Restrict the mocha dependency version to less than 6 and publish that as a patch version 
2. followed by  a major version bump to support mocha 6 and make it a peer dependency

if anyone has any other suggestions please let me know as ill start doing that now.

**Comment by jamie-sherriff** [comment]
Restrict the mocha dependency version to less than 6 and publish that as a patch version has been released as 2.5.2

**Comment by roymiloh** [comment]
Thanks, looks good.

**Comment by jamie-sherriff** [comment]
Mocha version 6 support is published under the tag next
`mocha-teamcity-reporter@next`
please let me know if you encounter any issues.

**Comment by njumbo** [comment]
Thanks @jamie-sherriff, we don't have a package-lock right now, this change will help us in the meantime. we will try it out today.

**Comment by jhdevuk** [comment]
@jamie-sherriff Amazing, thanks

**Comment by jamie-sherriff** [comment]
Fixed in 3.0.0 release
